openlibrary/plugins/openlibrary/code.py

```diff
@@ -787,7 +787,7 @@ def is_bot():
         'buzzbot', 'laserlikebot', 'baiduspider', 'bingbot',
         'mj12bot', 'yoozbotadsbot'
     ]
-    if 'HTTP_USER_AGENT' not in web.ctx.env:
+    if not web.ctx.env['HTTP_USER_AGENT']:
```

Comment:
I still think the best solution is the one previously described as its robust against both cases -- the case where `HTTP_USER_AGENT` is anomalously missing and the case where it is `None`.